File: 30 day code c++/day18.cpp
#include <iostream>
#include <queue>
#include <stack>

using namespace std;

class Solution {
  public:
    //Write your code here
    void pushCharacter(const char c_) { _s.push(c_); }
    void enqueueCharacter(const char c_) { _q.push(c_); }
    char popCharacter() {
        char c = _s.top();
        _s.pop();
        return c;
    }
    char dequeueCharacter() {
        char c = _q.front();
        _q.pop();
        return c;
    }
  private:
    queue<char> _q;
    stack<char> _s;
};

int main() {
    // read the string s.
    string s;
    getline(cin, s);
    
    // create the Solution class object p.
    Solution obj;
    
    // push/enqueue all the characters of string s to stack.
    for (int i = 0; i < s.length(); i++) {
        obj.pushCharacter(s[i]);
        obj.enqueueCharacter(s[i]);
    }
    
    bool isPalindrome = true;
    
    // pop the top character from stack.
    // dequeue the first character from queue.
    // compare both the characters.
    for (int i = 0; i < s.length() / 2; i++) {
        if (obj.popCharacter() != obj.dequeueCharacter()) {
            isPalindrome = false;
            
            break;
        }
    }
    
    // finally print whether string s is palindrome or not.
    if (isPalindrome) {
        cout << "The word, " << s << ", is a palindrome.";
    } else {
        cout << "The word, " << s << ", is not a palindrome.";
    }
    
    return 0;
}

File: 30daysOFcode_C#/day 18.cs
using System;
using System.Collections.Generic;
using System.IO;

class Solution {
        Queue<char> queue = new Queue<char>();

        Stack<char> stack = new Stack<char>();

        // Stack Methods
        public void pushCharacter(char ch)
        {
            stack.Push(ch);
        }

        public char popCharacter()
        {
            return stack.Pop();
        }

        // Queue Methods
        public void enqueueCharacter(char ch)
        {
            queue.Enqueue(ch);
        }

        public char dequeueCharacter()
        {
            return queue.Dequeue();
        }

        static void Main(String[] args) {
        // read the string s.
        string s = Console.ReadLine();

        // create the Solution class object p.
        Solution obj = new Solution();

        // push/enqueue all the characters of string s to stack.
        foreach (char c in s) {
            obj.pushCharacter(c);
            obj.enqueueCharacter(c);
        }

        bool isPalindrome = true;

        // pop the top character from stack.
        // dequeue the first character from queue.
        // compare both the characters.
        for (int i = 0; i < s.Length / 2; i++) {
            if (obj.popCharacter() != obj.dequeueCharacter()) {
                isPalindrome = false;

                break;
            }
        }

        // finally print whether string s is palindrome or not.
        if (isPalindrome) {
            Console.Write("The word, {0}, is a palindrome.", s);
        } else {
            Console.Write("The word, {0}, is not a palindrome.", s);
        }
    }
}